src/App.tsx

```diff
@@ -5,28 +5,51 @@ import * as RLP from "rlp";
 type AppState = {
   mode: "encoded" | "decoded";
   text: string;
-  rlp: any;
+  rlp: {
+    valid: boolean;
+    decoded: any;
+  };
 };
 
 class App extends React.Component<{}, AppState> {
   public constructor() {
     super({});
+
+    const stock = "0xc701c502c303c104";
     this.state = {
       mode: "encoded",
-      text: "0xc701c502c303c104",
-      rlp: [],
+      text: stock,
+      rlp: {
+        valid: true,
+        decoded: RLP.decode(stock),
```

Comment:
Why do you decode it on every change?
Why do not you show an error message when you click the decode button?

Reply:
RLP decoding is not expected to be a heavy task.
I wanted to give fast feedback to the user before clicking the button.